floodfill_algorithm/3dcube/flood_fill_3d.pyx

```diff
@@ -20,18 +20,17 @@ def cyfill(np.ndarray[np.float64_t, ndim=3] data, tuple start_coords, Py_ssize_t
     """
     cdef:
         Py_ssize_t x, y, z, xsize, ysize, zsize, orig_value
-        float thres 
+        float thres, orig_value_float
         set stack
     
     xsize = data.shape[0]
     ysize = data.shape[1]
     zsize = data.shape[2]
     
-    print data.shape[0]
-    print data[20, 20, 20]
-    orig_value = int(data[start_coords[0], start_coords[1], start_coords[2]])
-    # orig_value = data[20, 20, 20]
-	# this threshold should combine the relation between dragging distance and greyscale(0~255)
+    orig_value_float = data[start_coords[0], start_coords[1], start_coords[2]]
+    orig_value = int(orig_value_float)
```

Comment:
This was the main bug - you were comparing values below to orig_value which is an int - but we need to check compared to the true original value, so I've made it so we now have a float version of this variable. With this fix, our implementations agree.